tests/tests_default.yml

```diff
@@ -2,6 +2,54 @@
 ---
 - name: Ensure that the role runs with default parameters
   hosts: all
-  gather_facts: false  # test that role works in this case
-  roles:
-    - linux-system-roles.template
+  gather_facts: true  # test that role works in this case
+  become: yes
+  become_user: root
```

Comment:
will tests work without these two lines?

Reply:
> will tests work without these two lines?

Yes they will, and it is better not to put these in the playbook.  If you need to set `become`, `become_user`, etc. in your local testing, better to pass these on the `ansible-playbook` (or `tox -e qemu-ansible...`) command line:
```
ansible-playbook -h | grep -i become
                        [--become-method BECOME_METHOD]
                        [--become-user BECOME_USER]
                        [-K | --become-password-file BECOME_PASSWORD_FILE]
  --become-password-file BECOME_PASSWORD_FILE, --become-pass-file BECOME_PASSWORD_FILE
                        Become password file
  -K, --ask-become-pass
  control how and which user you become as on target hosts
  --become-method BECOME_METHOD
                        `ansible-doc -t become -l` to list valid choices.
  --become-user BECOME_USER
  -b, --become          run operations with become (does not imply password
```
so something like
```
tox -e qemu-ansible... -- --image-name centos-9 -b --become-user root -- tests/tests_default.yml
```
If you get tired of specifying these on the command line, you can put them in your `tests/ansible.cfg` - see https://docs.ansible.com/ansible/latest/reference_appendices/config.html

Reply:
> > will tests work without these two lines?
> 
> Yes they will, and it is better not to put these in the playbook. If you need to set `become`, `become_user`, etc. in your local testing, better to pass these on the `ansible-playbook` (or `tox -e qemu-ansible...`) command line:
> 
> ```
> ansible-playbook -h | grep -i become
>                         [--become-method BECOME_METHOD]
>                         [--become-user BECOME_USER]
>                         [-K | --become-password-file BECOME_PASSWORD_FILE]
>   --become-password-file BECOME_PASSWORD_FILE, --become-pass-file BECOME_PASSWORD_FILE
>                         Become password file
>   -K, --ask-become-pass
>   control how and which user you become as on target hosts
>   --become-method BECOME_METHOD
>                         `ansible-doc -t become -l` to list valid choices.
>   --become-user BECOME_USER
>   -b, --become          run operations with become (does not imply password
> ```
> 
> so something like
> 
> ```
> tox -e qemu-ansible... -- --image-name centos-9 -b --become-user root -- tests/tests_default.yml
> ```
> 
> If you get tired of specifying these on the command line, you can put them in your `tests/ansible.cfg` - see https://docs.ansible.com/ansible/latest/reference_appendices/config.html

. . . I meant for your local development - you should not commit `tests/ansible.cfg` to git.